File: src/main/java/net/hotamachisubaru/casino/Vault/Vault.java
package net.hotamachisubaru.casino.Vault;

import net.hotamachisubaru.casino.economy.EconomySetupHelper;
import net.milkbowl.vault.economy.Economy;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class Vault  {
    private static final String ECONOMY_PLUGIN_NAME = "Vault";
    private Economy economy = null;

    public Vault() {
        EconomySetupHelper setupHelper = new EconomySetupHelper();
        if (!setupHelper.setupEconomy()) {
            Bukkit.getLogger().warning("Vaultのセットアップに失敗しました。");
        } else {
            economy = setupHelper.getEconomy();
        }
    }

    public boolean has(Player player, double amount) {
        return isEconomySetup() && economy.has(player, amount);
    }

    public boolean withdraw(Player player, double amount) {
        if (has(player, amount)) {
            economy.withdrawPlayer(player, amount);
            return true;
        }
        return false;
    }

    public void deposit(Player player, double amount) {
        if (isEconomySetup()) {
            economy.depositPlayer(player, amount);
        }
    }

    public Economy getEconomy() {
        return economy;
    }

    private boolean isEconomySetup() {
        return economy != null;
    }
}

File: src/main/java/net/hotamachisubaru/casino/Vault/Vault.kt
package net.hotamachisubaru.casino.Vault

import net.milkbowl.vault.economy.Economy
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class Vault {
    var economy: Economy? = null
        private set

    init {
        if (!setupEconomy()) {
            Bukkit.getLogger().warning("Vaultのセットアップに失敗しました。")
        }
    }

    private fun setupEconomy(): Boolean {
        if (Bukkit.getServer().getPluginManager().getPlugin("Vault") == null) {
            Bukkit.getLogger().warning("Vaultプラグインが見つかりません。")
            return false
        }

        val rsp = Bukkit.getServer().getServicesManager().getRegistration<Economy?>(Economy::class.java)
        if (rsp == null) {
            Bukkit.getLogger().warning("Economyサービスが登録されていません。")
            return false
        }

        economy = rsp.getProvider()
        return economy != null
    }

    fun has(player: Player?, amount: Double): Boolean {
        return economy != null && economy!!.has(player, amount)
    }

    fun withdraw(player: Player?, amount: Double): Boolean {
        if (economy != null && economy!!.has(player, amount)) {
            economy!!.withdrawPlayer(player, amount)
            return true
        }
        return false
    }

    fun deposit(player: Player?, amount: Double) {
        if (economy != null) {
            economy!!.depositPlayer(player, amount)
        }
    }
}
